chips/sam4l/src/bscif.rs

```diff
@@ -344,3 +344,25 @@ pub unsafe fn enable_rc32k() {
         .rc32ktune
         .write(RC32kTuning::COARSE.val(0x1d) + RC32kTuning::FINE.val(0x15));
 }
+
+pub unsafe fn rc32k_enabled() -> bool {
+    return (*BSCIF).rc32kcr.is_set(RC32Control::EN);
+}
+
+pub unsafe fn setup_rc_1mhz() {
+    // Unlock the BSCIF::RC32KCR register
+    (*BSCIF)
+        .unlock
+        .write(Unlock::KEY.val(0xAA) + Unlock::ADDR.val(0x58));
+    (*BSCIF).rc1mcr.modify(RC1MClockConfig::CLKOEN::Output);
```

Comment:
Does this work? I recently (#929) found that the unlock on persists for one register action, and because `.modify()` is really a read then a write, the `.modify()` doesn't actually do anything.

Reply:
I've been skeptical that unlock actually does anything, since I had accidentally used the wrong unlock address for one of the clocks but the clock was still configured properly.

I just tried switching from the DFLL to the RC1M, with the unlock code commented out for the RC1M's setup and disable functions. It looks like the RC1M still works since I saw the UART's baud rate change from 115200 to 2400, which you'd expect going from a 48MHz to a 1MHz clock.